### PR TITLE
fix: Correctly encode user given content, such as passwords

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes https://github.com/stackabletech/nifi-operator/issues/624
Needs https://github.com/stackabletech/docker-images/pull/706

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/operator-binary/src/security/authentication.rs [hunk 1]
```diff
@@ -31,12 +27,15 @@ pub enum Error {
...
-    #[snafu(display(
-        "there was an error adding LDAP Volumes and VolumeMounts to the Pod and containers"
-    ))]
+    #[snafu(display("Failed to add LDAP volumes and volumeMounts to the Pod and containers"))]
     AddLdapVolumes {
         source: stackable_operator::commons::authentication::ldap::Error,
     },
+
+    #[snafu(display(
+        "The LDAP AuthenticationClass is missing the bind credentials. Currently nifi-operator only supports connecting to LDAP servers using bind credentials"
```
Comment (maltesander) [inline rust/operator-binary/src/security/authentication.rs:36]:
```suggestion
        "The LDAP AuthenticationClass is missing the bind credentials. Currently the NiFi operator only supports connecting to LDAP servers using bind credentials"
```

**Review by maltesander — DISMISSED** [review]
Very nice thanks! LGTM!

**Review by maltesander — APPROVED** [review]
LGTM when tests pass!

**Comment by sbernauer** [comment]
As the product images have been build now started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/57/console

**Comment by sbernauer** [comment]
Tests passed first try :rocket:

**Pull request merged**